test/util/numerics_test.py

```diff
@@ -216,5 +222,291 @@ def test_fast_1d_tensor_mult_error():
         fast_1d_tensor_mult(test_arr, [x, y, np.ones((4, 2))], (-2, -3, -4))
 
 
+# --- resize_array --- #
+
+
+dtype_params = ['float64', 'complex64', 'int64']
+dtype_ids = [' dtype = {} '.format(p) for p in dtype_params]
+
+
+@pytest.fixture(scope="module", ids=dtype_ids, params=dtype_params)
+def dtype(request):
+    return request.param
+
+
+pad_mode_params = list(_SUPPORTED_RESIZE_PAD_MODES)
+pad_mode_params.remove('constant')
+pad_mode_params.extend(['constant 0', 'constant 1'])
+pad_mode_ids = [" pad_mode = '{}' ".format(p) for p in pad_mode_params]
+
+
+@pytest.fixture(scope="module", ids=pad_mode_ids, params=pad_mode_params)
+def pad_mode(request):
+    return request.param
+
+
+variants = ['extend', 'restrict', 'mixed']
+setup_params = list(product(pad_mode_params, variants))
```

Comment:
It would be more clear what you meant if you instead passed the two fixtures as arguments to `resize_setup`. pytest supports "recursive" fixtures.